### PR TITLE
[LOOP-2998] user must completed the health access form

### PR DESCRIPTION
**Pull request by nhamming**
https://tidepool.atlassian.net/browse/LOOP-2998

As described in the ticket, one way to bypass the health access form is to press the home button when the health access form is presented.

### CODE REVIEW

File: Loop/View Controllers/StatusTableViewController.swift [hunk 1]
```diff
@@ -179,6 +180,18 @@ final class StatusTableViewController: LoopChartsTableViewController {
...
     }
+    
+    private func authorizeHealthStore(completion: @escaping (Bool) -> Void) {
+        deviceManager.authorizeHealthStore { accessFormWasCompleted in
+            // returned Bool only indicates if the user completed the health access form.
+            if accessFormWasCompleted {
+                completion(accessFormWasCompleted)
+            } else {
+                // if the user did not complete the health access form, present the health access form again so the user can allow or deny access
+                self.authorizeHealthStore(completion: completion)
```
Comment (darinkrauss) [inline Loop/View Controllers/StatusTableViewController.swift:191]: How would this happen? How could the form be displayed and dismissed, but not allowed or denied access?

**Review by darinkrauss — COMMENTED** [review]
Not sure I understand what is going on here.

File: Loop/View Controllers/StatusTableViewController.swift (hunk 1 above)
Comment (nhamming) [reply Loop/View Controllers/StatusTableViewController.swift:191]: There are steps to reproduce in the ticket, which include pressing the home button.

**Review by darinkrauss — APPROVED** [review]
LGTM! 🚀

**Pull request merged**